homeassistant/components/deconz/climate.py

```diff
@@ -0,0 +1,115 @@
+"""
+Support for deCONZ climate devices.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/climate.deconz/
+"""
+
+from homeassistant.components.climate import (
+    ClimateDevice, SUPPORT_ON_OFF, SUPPORT_TARGET_TEMPERATURE)
+from homeassistant.const import (
+    ATTR_BATTERY_LEVEL, ATTR_TEMPERATURE, TEMP_CELSIUS)
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .const import (
+    ATTR_ON, ATTR_SCHEDULER, CONF_ALLOW_CLIP_SENSOR, DOMAIN as DECONZ_DOMAIN)
+from .deconz_device import DeconzDevice
+
+DEPENDENCIES = ['deconz']
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the deCONZ climate devices.
+
+    Thermostats are based on the same device class as sensors in deCONZ.
+    """
+    gateway = hass.data[DECONZ_DOMAIN]
+
+    @callback
+    def async_add_climate(sensors):
+        """Add climate devices from deCONZ."""
+        from pydeconz.sensor import THERMOSTAT
+        entities = []
+        allow_clip_sensor = config_entry.data.get(CONF_ALLOW_CLIP_SENSOR, True)
+        for sensor in sensors:
+            if sensor.type in THERMOSTAT and \
+               not (not allow_clip_sensor and sensor.type.startswith('CLIP')):
+                entities.append(DeconzThermostat(sensor, gateway))
+        async_add_entities(entities, True)
+
+    gateway.listeners.append(
+        async_dispatcher_connect(hass, 'deconz_new_sensor', async_add_climate))
```

Comment:
I suggest making the signal name a constant.

Reply:
Better?

Reply:
Yes!